src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -94,6 +104,12 @@ protected override TypeInfo CreateTypeInfoCore()
                 return this;
             }
 
+            if (_isHiddenGlobalType)
+            {
+                _isCreated = true;
+                return null!;
```

Comment:
This is a bit sketchy, no? Are we sure users are going to know that calling this on the global type and no use the `null`?

Reply:
There was a discussion about this https://github.com/dotnet/runtime/issues/68840#issuecomment-1116726284, the global type only created internally (within `ModuleBuilder.CreateGlobalFunctions()`) and the returned null is not used.